holmes.go

```diff
@@ -355,7 +355,7 @@ func (h *Holmes) threadCheckAndShrink(threadNum int) {
 
 		h.logf("current thread number(%v) larger than threshold(%v), will start to shrink thread after %v", threadNum, opts.Threshold, opts.Delay)
 		time.AfterFunc(opts.Delay, func() {
-			h.startShrinkThread()
+			h.startShrinkThread(opts)
 		})
```

Comment:
`opts` is a pointer var, its scope is `threadCheckAndShrink`, it would be allocated on the heap if we pass it into another function `startShrinkThread`, which speeds up GC frequency. If we use a struct var(that's why we return a full copy of config), it allocated it on the stack and the memory used would be collected after the function life cycle. Correct me if i were wrong.

Reply:
这样传参确实会逃逸到堆上
除开逃逸的问题，这种改动，似乎也没啥好处？

Reply:
嗯，我当时是想减少锁占用，现在改成非指针了，应该不影响吧

Reply:
1. 一样的，闭包读取外部参数，就是会放堆上的
2. 这个锁优化其实没多少必要，本来就是很低频的读锁

Reply:
好滴，拿掉~